internal/worker/server.go

```diff
@@ -111,6 +112,30 @@ func (s *Server) VerifyIdentityHeader(nextHandler echo.HandlerFunc) echo.Handler
 	}
 }
 
+// This function should be started as a goroutine
+// Every 30 seconds it goes through all running jobs, removing any unresponsive ones.
+// It fails jobs which fail to check if they cancelled for more than 2 minutes.
+func (s *Server) WatchHeartbeats() {
+	//nolint:staticcheck // avoid SA1015, this is an endless function
+	for range time.Tick(time.Second * 30) {
+		for _, token := range s.jobs.Heartbeats(time.Second * 120) {
+			id, _ := s.jobs.IdFromToken(token)
+			log.Printf("Removing unresponsive job: %s\n", id)
+			result, err := json.Marshal(OSBuildJobResult{
+				Success: false,
+			})
+			if err != nil {
+				log.Printf("Error finishing unresponsive job: %v", err)
+				return
+			}
+			err = s.FinishJob(token, result)
```

Comment:
In `New()` you pass `nil` instead of `result` when you fail the job. Is that ok (i.e., will any parsers of this assume it is non-`nil`?), if so, maybe just do that here too?

Reply:
Ah yea, makes a lot more sense :)